### PR TITLE
fix: helm cache permissions

### PR DESCRIPTION
**Pull request by Brightside56**
This image is acting as base image for kargo, which runs as user with ID=1000. This PR is needed to create home directory maintain there permissions needed for helm cache

### CODE REVIEW

**Comment by krancour** [comment]
Here is the answer re: why Kargo switched to using `1000`:

https://github.com/akuity/kargo/pull/542#discussion_r1289532868

> It's possible that we don't need to change 65532 to 1000 here, but need to change 1000 back to 65532 there.

So we can change it here after all, although it may be moot if we determine that having a home dir doesn't actually improve anything. I am still hunting down answers on that.

**Comment by krancour** [comment]
I spent some time digging in today...

I can re-confirm now what I said before. The only time Kargo ever pulls charts is in the case where it runs a `helm dependency update`. (This would be a case where you're updating a so-called "umbrella chart" with a reference to an updated dependency. My instinct is that this isn't, or won't be, a very common use case.)

And in that case, a home directory at `/home/nonroot` won't accomplish anything, because all the promotion mechanisms do their work in a temporary directory and any commands they invoke set `HOME` to point to that temporary directory. This is for a variety of reasons, most of which have nothing to do with Helm, but WRT to Helm, I'd counted this as a good thing anyway because I wasn't confident that Helm could deal with concurrent updates to its cache without encountering race conditions.

Bottom line is that for this PR to have the intended effect, Kargo would need code changes to _not_ change the value `HOME` when invoking `helm dependency update`. That's a change we could entertain only after ascertaining that it is safe for multiple processes to concurrently update that cache. And even then, I would probably opt for that cache to be in a _different_ location as specified by the `XDG_CACHE_HOME` env var -- and I'd mount a persistent volume to that location to preserve the cache between controller restarts.

Since this PR doesn't achieve the intended effect, I'm going to go ahead and close this.

If I have somehow overlooked something, please do feel free to comment and re-open and I'd be happy to give this further consideration.

**Comment by Brightside56** [comment]
@krancour if I understand correctly, need to:
1. make sure that kargo/kargo-render caches will be not able to tamper each other when having common home dir
2. change permission here (in kargo-render repo) to be compatible with kargo
3. put this common home dir to persistent volume in kargo helm 

right?
